LIPs/LIP-livepeer_treasury.md

```diff
@@ -0,0 +1,108 @@
+
+Upon creation, LPT and other assets can be transfered into this smart contract via any mechanism.
+
+The mechanisms for making proposals via the `LivepeerGovernor`, and voting on proposals via the voting power and tallying mechanisms are described below.
+
+
+#### Governance over the treasury
+
+**Proposals**
+
+Proposals can be made by any user with a staked LPT balance exceeding the `Proposal Threshold`. Users will be able to submit text and media supported proposals, along with an amount of LPT to be released from the treasury to a specific address if the proposal passes.
+
+
+**Stake snapshotting**
+
+The spirit of Livepeer's existing delegated stake weighted voting for governance actions captured in [LIP-19](https://github.com/dob/LIPs/blob/dob/delta/LIPs/LIP-19.md) is maintained in this LIP for treasury management. Orchestrators can vote on proposals carrying the full weight of their delegated stake, however any delegator can show up to override the vote of their orchestrator on behalf of their own stake. However, there are some differences in the mechanics in order to support the Governor framework. The largest is that:
+
+* **Stake amounts must be snapshotted at the start round of voting on a proposal**, whereas in existing Livepeer governance stake amounts are tallied at the conclusion of a proposal.
+
+This update requires a new stake snapshotting library in the Livepeer protocol smart contracts, and a number of hooks in stake related actions within the existing BondingManager. 
+
+
+**Voting**
+
+When proposals are made on chain, they are introduced with a `Voting Delay` and a `Voting Period`. This LIP proposes the initial values of these as 1 round and 10 rounds respectively. After the `Voting Delay` has passed, delegators and orchestrators can vote for, against, or abstain on proposals until the `Voting Period` has ended. Abstained votes will count towards qorum, but will not affect the for or against tallies.
+
+* The `QUORUM` and `QUOTA` values from Livepeer's existing governance will be used to determine whether the proposal has received enough votes to be valid, and if the poll passed or failed. At the time of writing, then `QUORUM` value is 33% of active stake, and the `QUOTA` is 50%, meaning that as long as 1/3rd of active stake votes, if the majority of the votes are in favor passing, the proposal will pass.
+
+**Execution**
+
+Upon the end of the `Voting Period`, it will be determinable on chain whether the proposal passed or did not pass. If it passed, then there is an `execute` transaction callable by anyone, that will execute the associated transaction associated with the proposal. By default, initially the only transaction type will be to send LPT from the treasury to the specified receive address. If the proposal did not pass or did not reach quorum, then the `execute` function will revert and no action will be taken.
```

Comment:
@victorges Yes I was thinking about this as well. When proposals are submitted, do we have the ability to enforce that the only txn type is a transfer of LPT from the treasury to a receive address? In the future we can update this to enable other types of governance actions?

Reply:
That would be possible to be implemented if we override the `Governor.propose` function and make a few checks. The only change would be that we probably want to allow any of the "governance" proposals as well (e.g. updating voting delay, period, timelock delay, etc).

The implementation can have a list of allowed `calldatas / targets`. Would need to add this to the spec and implementation though, which might be a 0.5-1 day effort. Let me know if we do want this behavior and I can give it a shot on https://github.com/livepeer/protocol/pull/615 !

Reply:
@victorges Yes I think this is a good idea. But I would like to better understand the process for updating this overrided Governor.propose function in the future to allow greater flexibility in what can be done via governance actions. Is it upgradeable...and if so who has permission to upgrade it? I guess this would be solved if the valid Governor.propose actions are:

* Token transfers from the treasury
* Updates to the governance parameters
* Updates to the function that performs these checks...so that a different one can be selected in the future?